packages/@glimmer/syntax/lib/generation/print.ts

```diff
@@ -57,7 +57,7 @@ export default function build(ast) {
     }
     break;
     case 'MustacheCommentStatement': {
-      output.push(compactJoin(['{{!', ast.value, '}}']));
+      output.push(compactJoin(['{{!--', ast.value, '--}}']));
```

Comment:
Besides, the opposite problem of being "inaccurate" also currently exists since all `{{!--` style comments get printed as `{{!`. So the only thing this change gains is getting rid of a parse error and it loses nothing.